mailpoet/lib/Cron/ActionScheduler/Actions/DaemonRun.php

```diff
@@ -79,12 +79,11 @@ public function getDaemonExecutionLimit(): float {
    */
   public function afterProcess(): void {
     if ($this->wordpressTrigger->checkExecutionRequirements()) {
+      $this->actionScheduler->scheduleImmediateSingleAction(self::NAME);
       // The automatic rescheduling schedules the next recurring action to run after 1 second.
       // So we need to wait before we trigger new remote executor to avoid skipping the action
```

Comment:
I guess these comments are not relevant anymore? Even the sleep may not be needed, but I don't know, is it better to keep it for some reason? I guess it may be safer to keep it.

Reply:
I agree it is safer to keep it. There is a similar [sleep in the action scheduler async runner](https://github.com/woocommerce/action-scheduler/blob/trunk/classes/ActionScheduler_AsyncRequest_QueueRunner.php#L91-L96). I'm not really sure why and how it helps, though.